### ISSUE TITLE
Crashed Sapper-Server causing weird system-wide DNS-problems

### ISSUE THREAD

**Issue by tippexx** [sveltejs/sapper#1185]
**Describe the bug**
I experienced system-wide problems surfing the web. I could only access certain domains, on others I would get "page can't be reached" but a few moments later that exact same page could be reached and the ones I was able to access before couldn't anymore. I tried different browsers, same thing. The domains were well known (i.e google, github, duckduckgo etc..). If a domain was accessible or not could change in a matter of seconds (i.e. after 4 sec. the page would load but then other previously working domains couldn't be accessed anymore). I could still successfully ping the unreachable domains. Restarting my PC would fix the problem for a couple of minutes (~30min) before the above mentioned behavior reoccurred.

**To Reproduce**
Initially:
Using the unchanged template, I started a dev server (`npm run dev`) and also built the project (`npm run build`). I then moved the build folder outside the `__sapper__` folder on the `Desktop` and tried to run it using `node ./build`. It didn't run because it was missing dependencies. I then tried again from the  `__sapper__` folder inside the project and it ran (`node __sapper__/build`). 

Later (I don't remember if it was right away or after some time) I experienced the above mentioned problems.

Now:
For me every crached server.js cases the problem.

**Information about your Sapper Installation:**
- Browser and the version: (Version 81.0.4044.138 (Official Build) (64-bit), Firefox 76.0.1 (64-bit), Edge Version 81.0.416.72 (Official build) (64-bit))

- Operating system: (Windows 10)

- Hosting environment: (Local)

- Sapper version (0.27)

- Svelte version (3)

- Application uses Rollup

- Node v12.16.2

**Severity**
Every server.js crash ruins my network adapter and I have to reinstall the driver to get it back working properly. I can't work like this as I need to be able to use the Internet. 

Blocking for some users.

**Additional context**
I did not have any problems on my network just my PC. The problem was also independent of connection (wireless or wired) and DNS-Server.

**Recent findings and speculation**
Rerunning `npm run dev` and `^C` before it crashed again seems to make the unreachable domains reload and working again (Not always though. You need to get the right timing - it may also be a coincident). But does not resolve the Problem as a whole.

Given that **reinstalling the network adapter driver fixed** it as a whole and a crashed local server is causing it, I would guess that it has something to do with the environment variables that are change or set up to get it working. Maybe something regarding the live reloading?

If something is unclear or there is a need for more information please tell me and I will try my best to clarify.

**Comment by dmgolembiowski** [comment]
Hi @tippexx, I believe you; I, too, was experiencing browser issues like these when running sapper lately, but never thought it was anything but my F5 load balancer. What's more, I experienced highly unusual zombie port blockages that could not be tracked down (on Debian) until I finally gave in and restarted the server. Moreover, when I went back and tried changing the `PORT` values in one of the  npm run scripts, things would temporarily work until (as you said) "_[r]erunning npm run dev and ^C before it crashed again seems to make the unreachable domains reload_". 

Can you include the dependencies, devDependencies, and run scripts from your package.json file for this project?

**Comment by jbelien** [comment]
Same issue today, running `npm run dev` on WSL on Windows 10.

Once the server is crashed, I can't access any website anymore.

**Comment by tippexx** [comment]
@dmgolembiowski 

These should be the dependencies:

> {
  "name": "TODO",
  "description": "TODO",
  "version": "0.0.1",
  "scripts": {
    "dev": "sapper dev",
    "build": "sapper build",
    "export": "sapper export",
    "start": "node __sapper__/build",
    "cy:run": "cypress run",
    "cy:open": "cypress open",
    "test": "run-p --race dev cy:run"
  },
  "dependencies": {
    "compression": "^1.7.1",
    "polka": "next",
    "sirv": "^0.4.0"
  },
  "devDependencies": {
    "npm-run-all": "^4.1.5",
    "sapper": "^0.27.0",
    "svelte": "^3.0.0",
    "svelte-loader": "^2.9.0",
    "webpack": "^4.7.0"
  }
}

What I did not state is that I was using WSL (Ubuntu) for Windows 10 too.
Also it seems to have been a webpack project. This does not matter tho. I had it reoccur using rollup as well.

@jbelien For me it happens not every time the server crashes. It does not occur if the crash is due to an error in one of the .svelte route files for example. hbu? 

How to reproduce easily: 
i.e: Make an non existing `import ` (or anything else that will error out) in server.js, run `npm run dev` and that's how to recreate 100% for me.

**Comment by tippexx** [comment]
Also, crashing the production (not dev) server will cause the problem too. (I achieved that through moving the `__sapper__` folder out of the project root folder, as mentioned) 

Due to that I would believe, that it has nothing to do w/ livereload @dmgolembiowski , unless livereload is still a dependency of the production build (not sure about that).

**Comment by jbelien** [comment]
> @jbelien For me it happens not every time the server crashes. It does not occur if the crash is due to an error in one of the .svelte route files for example. hbu?
>
> How to reproduce easily:
> i.e: Make an non existing import (or anything else that will error out) in server.js, run npm run dev and that's how to recreate 100% for me.

Yes, you're probably right. My crash was due to a failed import !

**Comment by dmgolembiowski** [comment]
@jbelien Cool, thanks for the insight.

**Comment by antony** [comment]
Did you @dmgolembiowski @tippexx get any further with diagnosing this?

**Comment by tippexx** [comment]
@antony, not yet. But I just thought it is worth it to try recreate this using expressJS, this may be an issue with polka. No promises but I may have the time to play around with it this weekend.

**Comment by babeard** [comment]
I too have been having this problem multiple times a day for a long while now. (Windows 10 / WSL2 Ubuntu / Polka/ Rollup).

Of course a fix would be great but in the meantime I have been killing any processes named `init` and then getting back into the terminal and `npm run dev` rather than to restarting the whole computer. 

![Untitled-1](https://user-images.githubusercontent.com/2446647/89650688-6cc19800-d888-11ea-86fd-9231b4bfbe62.jpg)

#### Steps to reproduce for me
1) Put a .js file in the routes folder with the following contents
```javascript
// src/routes/hello.js

import hi from "non-exists";

throw new Error(hi);
```
2) `npm run dev`
3) Open incognito window, so you don' have any cached DNS, and try to go to any website.

Interestingly, this only happens for me using the default rollup template but not the default webpack one. I am curious if anyone else has found this to be true as well.

**Comment by benmccann** [comment]
Any bug would most likely be in a library used by Sapper and not in Sapper itself. Some other things worth trying would be testing Express and testing various versions of Polka (0.41, 1.0, etc.)

**Comment by babeard** [comment]
Here are some configurations I've been able to test so far.

|       | version | has issue|
|-------|---------|-------|
| polka | 0.4.0  | yes   |
|       | 0.5.0  | yes   |
|       | 0.5.1  | yes   |
|       | 0.5.2  | yes   |
|       | 1.0.0-next.11  | yes   |
| express| 3.21.2  | yes   |
|       | 4.17.1  | yes   |
| sapper (w/polka) | ^0.27.0  | yes   |
|   | ^0.26.0  | yes   |
| node | 12.18.2  | yes   |
|  | 14.5.0  | yes   |
